### DISCUSSION TITLE
Why is the IPv6 used address bind to 0.0.0.0?

### DISCUSSION THREAD

**Discussion by bitcoffeeiux** [fortio/fortio#919] [Q&A] [answered]
When I use fortio bind to 0.0.0.0:8080, the port is listening to tcp6. Why? I just want it to bind all the ipv4 network cards on the local machine. What should I do?

[root@k8s-node2 ~]# fortio server -http-port 0.0.0.0:8080
11:15:12 Fortio 1.38.1 tcp-echo server listening on tcp [::]:8078
11:15:12 Fortio 1.38.1 udp-echo server listening on udp [::]:8078
11:15:12 Fortio 1.38.1 grpc 'ping' server listening on tcp [::]:8079
11:15:12 Fortio 1.38.1 https redirector server listening on tcp [::]:8081
11:15:12 Fortio 1.38.1 http-echo server listening on tcp [::]:8080
11:15:12 Data directory is /home/liuxin/code/open_kernel
11:15:12 REST API on /fortio/rest/run, /fortio/rest/status, /fortio/rest/stop
         UI started - visit:
                http://[::]:8080/fortio/
11:15:12 I fortio_main.go:285> Note: not using dynamic flag watching (use -config to set watch directory)
11:15:12 I fortio_main.go:293> All fortio 1.38.1 h1:WZZPHKQUCwfYFSHdWrUNsSIVyEwT4NFiCCyAETP6mGY= go1.19.2 amd64 linux servers started!

[root@k8s-node2 ~]# netstat -anotup |grep 8080
tcp6       0      0 :::8080                 :::*                    LISTEN      3377216/fortio       off (0.00/0/0)

**Comment by ldemailly** [comment, marked as answer]
it's a bit of a go question, and I don't know the answer, outside of passing the ipv4 you want to bind to?

what is the issue with having it listen to all?

**Reply by ldemailly** [reply]
I guess it would need to be "tcp4" on https://github.com/fortio/fortio/blob/master/fnet/network.go#L137

I could make a flag but I am curious
